src/app/+step1/step1.component.html

```diff
@@ -1,9 +1,19 @@
 <p>This is step1.</p>
 
   <form [ngFormModel]="step1">
-    <form-field [field]="fm.getField('input_name')"></form-field>
+    <form-field [field]="fm.getField('input_name')" 
+     (value)="update($event.field, $event.value)"></form-field>
     
-    <form-field [field]="fm.getField('input_age_user')"></form-field>
+    <form-field [field]="fm.getField('input_age_user')"
+      (value)="update($event.field, $event.value)"></form-field>
     
-    <form-field [field]="fm.getField('input_relationship_status')"></form-field>
+    <form-field [field]="fm.getField('input_relationship_status')"
+      (value)="update($event.field, $event.value)"></form-field>
+    
+    <form-field [field]="fm.getField('input_region')"
+    (value)="updateLocationFied($event)"></form-field>
+    
+    <form-field [field]="fm.getField('input_area')"
+    (value)="update($event.field, $event.value)" [ngModel]="selectedArea"></form-field>  
```

Comment:
these custom fields also support ngModel so their value can be externally controlled, such as in this example